modules/reference/pages/data-transform-rust-sdk.adoc

```diff
@@ -0,0 +1,399 @@
+= Rust SDK for Data Transforms
+:description: Work with data transforms using Rust.
+
+Rust SDK reference documentation for Redpanda data transforms.
+
+The `redpanda_transform_sdk` crate contains the core functions and types for transforming data within Redpanda.
+
+== Functions
+
+The Rust SDK includes the following https://doc.rust-lang.org/rust-by-example/fn.html[function] to manage data transforms to an output topic.  
+
+=== on_record_written
+
+The `on_record_written()` function registers a callback that transforms records after they have been written to an input topic. This function is executed when a record is written to the topic. The callback is executed after the record has been written and fsynced to disk and the producer has been acknowledged.
+
+[,rust]
+----
+pub fn on_record_written<E, F>(cb: F) -> !
+where
+    E: Debug,
+    F: Fn(WriteEvent<'_>, &mut dyn RecordWriter) -> Result<(), E>,
+----
+
+Callback parameters
+
+- <<writeevent,`WriteEvent`>>
+- <<recordwriter, `RecordWriter`>>
+
+Example:
+
+This transform function copies the same data from an input topic to an output topic.:
+
+[,rust]
+----
+use anyhow::Result;
+use redpanda_transform_sdk::*;
+
+fn main() {
+    on_record_written(my_transform);
+}
+
+// This will be called for each record in the source topic. The output records returned will be written to the destination topic.
+fn my_transform(event: WriteEvent, writer: &mut RecordWriter) -> Result<()> {
+    writer.write(event.record)?;
+    Ok(())
+----
+
+`on_record_written()` should be called from the `main()` function, since the method blocks and runs forever unless explicitly stopped. Carry out any necessary pre-processing steps before calling this method.
+
+If an error is returned, Redpanda prints the error and aborts processing in the VM. Redpanda will then retry processing with backoff.
+
+---
+
+== Structs
+
+A https://doc.rust-lang.org/rust-by-example/custom_types/structs.html[struct] is a custom data type that allows you to define a structure composed of related data attributes.
+
+=== BorrowedHeader
+
+`BorrowedHeader` is a zero-copy <<borrowedrecord,`BorrowedRecord`>> header.
+
+[,rust]
+----
+pub struct BorrowedHeader<'a> {
+    key: &'a [u8],
+    value: Option<&'a [u8]>,
```

Comment:
Let's remove the insides of these, they are likely to go out of date and they are private.